core/src/main/java/org/apache/metamodel/create/AbstractTableCreationBuilder.java

```diff
@@ -123,9 +123,22 @@ public String toSql() {
                     && !column.isNullable().booleanValue()) {
                 sb.append(" NOT NULL");
             }
-            if (column.isPrimaryKey()) {
-                sb.append(" PRIMARY KEY");
-            }
+        }
+        boolean primaryKeyExists = false;
+        for(int i = 0 ; i < columns.length ; i++) {
+        	if(columns[i].isPrimaryKey()) {
+        		if(!primaryKeyExists) {
```

Comment:
Oh I see that the original file is actually using tabs ... strange. I guess we should reformat the whole file then :-/